`TestReactorEventuallyRemovesExpiredTransaction` flake

## Problem

Test flake

```
--- FAIL: TestReactorEventuallyRemovesExpiredTransaction (0.20s)
    reactor_test.go:205: 
        	Error Trace:	/home/runner/work/celestia-core/celestia-core/mempool/cat/reactor_test.go:205
        	Error:      	Should be false
        	Test:       	TestReactorEventuallyRemovesExpiredTransaction
FAIL
```

## Occurences
1. https://github.com/celestiaorg/celestia-core/actions/runs/6694512516/job/18259427654?pr=1128#step:6:121

--- a/mempool/cat/reactor_test.go
+++ b/mempool/cat/reactor_test.go
@@ -239,8 +239,11 @@ func TestReactorEventuallyRemovesExpiredTransaction(t *testing.T) {
 	require.True(t, reactor.mempool.Has(key))
 
 	// wait for the transaction to expire
-	time.Sleep(reactor.mempool.config.TTLDuration * 2)
-	require.False(t, reactor.mempool.Has(key))
+	require.Eventually(t,
+		func() bool { return !reactor.mempool.Has(key) },
+		4*reactor.mempool.config.TTLDuration,
+		50*time.Millisecond,
+		"transaction was not removed after TTL expired")
 }
 
 func TestLegacyReactorReceiveBasic(t *testing.T) {

